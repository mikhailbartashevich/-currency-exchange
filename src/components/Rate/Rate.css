.Rate {
  display: flex;
  color: #446dd2;
  background: white;
  border: 1px solid #2c293336;
  border-radius: 10px;
  width: 140px;
  height: 20px;
  justify-content: center;
  align-items: center;
}

.Rate_label {
  margin-right: 5px;
}
