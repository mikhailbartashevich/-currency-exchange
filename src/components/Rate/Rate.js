import React from 'react';
import './Rate.css';

export function Rate(props) {
  return (
    <div className="Rate">
      <span>1&nbsp;{props.inputCurrency.symbol}&nbsp;=&nbsp;</span>
      <span className="Rate_value">{props.rate}</span>
      <span className="">&nbsp;{props.outputCurrency.symbol}</span>
    </div>
  );
}
