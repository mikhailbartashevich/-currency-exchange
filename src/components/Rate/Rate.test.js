import React from 'react';
import { shallow } from 'enzyme';
import { Rate } from './Rate';

describe('ExchangeFab', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Rate debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<Rate />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const inputCurrency = { currency: 'USD', symbol: '$' };
    const outputCurrency = { currency: 'EUR', symbol: 'â‚¬' };
    const component = shallow(
      <Rate
        rate="0.23"
        inputCurrency={inputCurrency}
        outputCurrency={outputCurrency}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
