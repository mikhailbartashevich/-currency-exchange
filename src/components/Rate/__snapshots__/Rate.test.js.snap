// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeFab should render correctly in "debug" mode 1`] = `
<RateRaw
  debug={true}
/>
`;

exports[`ExchangeFab should render correctly with no props 1`] = `<RateRaw />`;

exports[`ExchangeFab should render correctly with props 1`] = `
<RateRaw
  inputCurrency={
    Object {
      "currency": "USD",
      "symbol": "$",
    }
  }
  outputCurrency={
    Object {
      "currency": "EUR",
      "symbol": "â‚¬",
    }
  }
  rate="0.23"
/>
`;
